app/scenes/channel/channel_header/channel_header_container.js

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {connect} from 'react-redux';
+
+import {Constants} from 'service/constants';
+import {getMyPreferences, getTheme} from 'service/selectors/entities/preferences';
+import {getCurrentUserId, getUser} from 'service/selectors/entities/users';
+import {getUserIdFromChannelName} from 'service/utils/channel_utils';
+import {displayUsername} from 'service/utils/user_utils';
+
+import ChannelHeader from './channel_header';
+
+function mapStateToProps(state, ownProps) {
+    const currentChannel = ownProps.currentChannel;
+
+    let displayName = '';
+    if (currentChannel) {
+        if (currentChannel.type === Constants.DM_CHANNEL) {
+            const otherUser = getUser(state, getUserIdFromChannelName(getCurrentUserId(state), currentChannel));
+
+            if (otherUser) {
+                displayName = displayUsername(otherUser, getMyPreferences(state));
+            }
+        } else {
+            displayName = currentChannel.display_name;
+        }
+    }
+
+    return {
```

Comment:
I'm kind of amazed that this works as it is. And I'm 0/5 on the name. Drawer probably makes more sense if we just had the RN app, but since it mirrors the web app, it would also make sense to keep the components with the same name